tyr-runner/src/main/java/org/jboss/tyr/check/DescriptionCheck.java

```diff
@@ -25,21 +26,24 @@
 import java.util.Scanner;
 import java.util.regex.Matcher;
 
-public class RequiredRowsCheck implements Check {
+public class DescriptionCheck implements Check {
 
     private List<RegexDefinition> rows;
+    private List<OptionalPatterns> optionalPatterns;
 
-    public RequiredRowsCheck(List<RegexDefinition> rows) {
+    public DescriptionCheck(List<RegexDefinition> rows, List<OptionalPatterns> optionalPatterns) {
         this.rows = rows;
+        this.optionalPatterns = optionalPatterns;
     }
 
     @Override
     public String check(JsonObject payload) {
         List<RegexDefinition> requiredRows = new ArrayList<>(rows);
+        List<OptionalPatterns> optionalRows = new ArrayList<>(optionalPatterns);
         String description = payload.getJsonObject(Utils.PULL_REQUEST).getString(Utils.BODY);
 
         try (Scanner scanner = new Scanner(description)) {
-            while (scanner.hasNextLine() && !requiredRows.isEmpty()) {
+            while (scanner.hasNextLine() || (!requiredRows.isEmpty() && !optionalRows.isEmpty())) {
```

Comment:
This doesn't seem correct. You want to iterate until there is still check in *either* requiredRows OR optionalRows.